### DISCUSSION TITLE
Security concerns for a peace of mind

### DISCUSSION THREAD

**Discussion by ioucyf** [supabase/supabase#11343] [Questions] [answered]
**Is it safe for the users when I implement supabase auth (and other services after authentication) in the client with anon key?** 

I've been reading a lot of articles (and videos) regarding auth implementations and most would advise to not use local storage for storing user sessions or JWT tokens, and even better to not use JWT at all and always use cookies with secure and httpOnly.

The best overall experience I can get, by a huge margin, is by implementing supabase in the client just as the docs suggest it; _and that's how I'd love to use it as well._

The team is phenomenal at what the do and the features they continue to offer. They obviously know their stuff. But I just can't help wondering: what am I missing? 

Please help me understand.

**Comment by kiwicopple** [comment, marked as answer]
> in the client with anon key?

Yes, but _only if you turn on Row Level Security and you have RLS Policies_. 

> I just can't help wondering: what am I missing?

You're not missing anything - you still need to implement Authorization rules, it's just that Supabase/Postgres gives you an option to implement those rules in your database. This is optional - if you don't want to use this approach, and prefer to write AuthZ in your middleware then that's also an option.

**Reply by ioucyf** [reply]
Thank you so much for the answer! 🙏🏼
